torch/csrc/utils/python_numbers.h

```diff
@@ -100,30 +106,28 @@ inline bool THPUtils_unpackBool(PyObject* obj) {
 }
 
 inline bool THPUtils_checkDouble(PyObject* obj) {
-  bool is_numpy_scalar;
 #ifdef USE_NUMPY
-  is_numpy_scalar = torch::utils::is_numpy_scalar(obj);
-#else
-  is_numpy_scalar = false;
+  if (torch::utils::is_numpy_scalar(obj) && ! PyBool_Check(obj)) {
```

Comment:
Why did this change?

Reply:
On master,
```
tensor = torch.zeros(3)
tensor.bernoulli_(p=True)
```
works. `p` is a `float` argument. I'm not sure if this is intentional or not, but the check against PyBool here would make it so that `tensor.bernoulli_(p=True)` errors out.

Reply:
I think you're right it's probably not needed and so I'll remove it, but I think it would only potentially affect something like `tensor.bernoulli_(p=np.array([True])[0])` which doesn't work on master either.

Reply:
It doesn't affect `tensor.bernoulli_(p=np.array([True])[0])` -- `PyBool_Check` returns False for np.bool_ so having it there is redundant.

Reply:
right (and removed)